src/application/models/v2/Appointments_model_v2.php

```diff
@@ -340,14 +345,19 @@ public function getAllAppointmentBy($service, $aggregates = FALSE, $otherRequest
                 $this->db->where('id_integrated', $otherQuery);
             }
         }
-
+       
         switch ($type) {
             case self::CUSTOMER:
                 $condition['id_users_customer'] = $service['id'];
                 break;
             case self::SERVICE:
                 $condition['id_services'] = $service[0]->id;
                 break;
+            case self::CUSTOMER_SERVICE:
+                // both customer and service
+                $condition['id_users_provider'] =$this->user_model_v2->find_by_id_integrated($id_user_integrated)['id'];
+                $condition['id_services'] = $this->services_model_v2->find_by_id_integrated($id_service_integrated)[0]->id;
```

Comment:
What problem if the service associating to id_service_integrated is not existing ?

I expect it failure somewhere with a clear error instead of array accessing error

Reply:
$this->user_model_v2->find_by_id_integrated already handle not found exception, in case one of them not exists in DB, definitely 404 will be thrown with detail message

Reply:
OK